app_feup/lib/model/entities/exam.dart

```diff
@@ -45,38 +47,42 @@ class Exam {
   String year;
   DateTime date;
 
-  Exam.secConstructor(String subject, String begin, String end, String rooms,
-      String day, String examType, String weekDay, String month, String year) {
+  Exam.secConstructor(String subject, DateTime beginDateTime, DateTime endDateTime, String rooms,
+      String examType, String weekDay) {
     this.subject = subject;
-    this.begin = begin;
-    this.end = end;
+    this.beginDateTime = beginDateTime;
+    this.endDateTime = endDateTime;
+    this.begin = beginDateTime.hour.toString()+':'+beginDateTime.minute.toString();
+    this.end = endDateTime.hour.toString() + ':' + endDateTime.minute.toString();
     this.rooms = rooms.split(',');
-    this.day = day;
     this.examType = examType;
     this.weekDay = weekDay;
-    this.month = month;
-    this.year = year;
+    this.day = beginDateTime.day.toString();
+    this.month = months.keys
+        .firstWhere((k) => months[k] == this.beginDateTime.month, orElse: () => null);
 
-    final monthKey = months[this.month];
-    this.date = DateTime.parse(year + '-' + monthKey + '-' + day);
+    this.year = beginDateTime.year.toString();
+    this.date = DateTime(beginDateTime.year,beginDateTime.month,beginDateTime.day);
   }
 
   Exam(String schedule, String subject, String rooms, String date,
       String examType, String weekDay) {
-    this.subject = subject;
-    this.date = DateTime.parse(date);
     final scheduling = schedule.split('-');
     final dateSepared = date.split('-');
```

Comment:
`splittedDate` is a better name for this variable, dont you think?

Reply:
Yes, I agree! `dateSepared` was the string name in the old code, but I think your suggestion would fit better.